sklearn/metrics/_classification.py

```diff
@@ -141,6 +145,28 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _nan_average(scores, weights):
+    """
+    Wrapper for np.average, with np.nan values being ignored from the average
+    This is similar to np.nanmean, but allowing to pass weights as in np.average
+    """
+
+    mask = np.isnan(scores)
+    if mask.all():
+        return np.nan
+    if weights is None:
+        return np.average(scores[~mask])
+    if isinstance(weights, list):
+        weights = np.array(weights)
```

Comment:
Nit: This will handle the case for ndarray and list. For ndarray, there is no copy.

```suggestion
    weights = np.asarray(weights)
```